tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Search.cs

```diff
@@ -956,10 +956,10 @@ internal bool UnPinThisRecord(ClientContext clientContext, Client client, object
                             string userPinnedDocument = !string.IsNullOrEmpty(Convert.ToString(listItems[0][userPinnedDetails.PinnedListColumnDetails],
                                 CultureInfo.InvariantCulture)) ? Convert.ToString(listItems[0][userPinnedDetails.PinnedListColumnDetails],
                                 CultureInfo.InvariantCulture) : string.Empty;
-                            Dictionary<string, DocumentData> userpinnedDocumentCollection =
-                                JsonConvert.DeserializeObject<Dictionary<string, DocumentData>>(userPinnedDocument);
-                            if (!string.IsNullOrWhiteSpace(userPinnedDetails.URL) && userpinnedDocumentCollection.ContainsKey(WebUtility.HtmlEncode(userPinnedDetails.URL))
-                                    || !string.IsNullOrWhiteSpace(userPinnedDetails.URL) && userpinnedDocumentCollection.ContainsKey(userPinnedDetails.URL)
+                            Dictionary<string, DocumentData> userpinnedDocumentCollection = new Dictionary<string, DocumentData>(StringComparer.InvariantCultureIgnoreCase);
```

Comment:
@WilsonReddy Fixed the comparison logic for pin and unpin functionality. Implemented case insensitive comparison